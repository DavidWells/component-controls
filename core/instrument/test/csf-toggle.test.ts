import { loadStoriesTests } from './loadTestFiles';

describe('csf-toggle', () => {
  loadStoriesTests('csf', 'toggle');
});
