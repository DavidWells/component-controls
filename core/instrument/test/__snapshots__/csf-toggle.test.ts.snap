// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csf-toggle toggle-story.js 1`] = `
Object {
  "components": Object {
    "a7e5ad523f225128d8b53302689e5a27": Object {
      "from": "./Toggle",
      "importedName": "Toggle",
      "imports": undefined,
      "loc": undefined,
      "name": "Toggle",
      "request": undefined,
      "source": undefined,
    },
  },
  "doc": Object {
    "component": "Toggle",
    "components": Object {},
    "componentsLookup": Object {
      "Toggle": "a7e5ad523f225128d8b53302689e5a27",
    },
    "date": 2020-03-07T04:02:17.329Z,
    "dateModified": 2020-03-07T04:02:32.477Z,
    "package": "907f6c0f914114634c0f7069bb74b196",
    "title": "Components/Toggle",
  },
  "packages": Object {
    "907f6c0f914114634c0f7069bb74b196": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.5.7",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "907f6c0f914114634c0f7069bb74b196",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/toggle/toggle-story.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.5.7",
    },
  },
  "stories": Object {
    "allToggles": Object {
      "arguments": Array [],
      "id": "allToggles",
      "loc": Object {
        "end": Object {
          "column": 1,
          "line": 28,
        },
        "start": Object {
          "column": 26,
          "line": 9,
        },
      },
      "name": "allToggles",
      "source": "() => {
  const [checked, setChecked] = React.useState(false);
  return (
    <div>
      <p>Default toggle</p>
      <Toggle checked={checked} onChange={check => setChecked(check)} />
      <br />
      <p>Custom labels</p>
      <Toggle
        checked={checked}
        labels={{
          true: 'YES',
          false: 'NO!',
        }}
        onChange={check => setChecked(check)}
      />
      <br />
    </div>
  );
}",
    },
  },
  "transformed": "import React from 'react';
import { Toggle } from './Toggle';

export default {
  title: 'Components/Toggle',
  component: Toggle,
};

export const allToggles = () => {
  const [checked, setChecked] = React.useState(false);
  return (
    <div>
      <p>Default toggle</p>
      <Toggle checked={checked} onChange={check => setChecked(check)} />
      <br />
      <p>Custom labels</p>
      <Toggle
        checked={checked}
        labels={{
          true: 'YES',
          false: 'NO!',
        }}
        onChange={check => setChecked(check)}
      />
      <br />
    </div>
  );
};
",
}
`;
